src/Models/WpComposite.php

```diff
@@ -107,6 +107,24 @@ private static function register_permalink() {
                     $request->query_vars['pagename'] = $request->query_vars['category_name'];
                     unset( $request->query_vars['category_name'] );
                 }
+                /*
+                 * The above page check would have been applied and unset the category name, therefore we need to check again to avoid unwanted
+                 * undefined index errors.
+                 * The bellow 'hack' will be applied to category pages to make sure we can't access sub-category URL directly without the parent-category
+                 * E.g:
+                 * - http://gds.dev/terrasse/ (terrasse is the parent). Url works
+                 * - http://gds.dev/terrasse/fliseterrasse/ (fliseterrasse sub-category). Url works
+                 * - http://gds.dev/fliseterrasse/ (This should not work and throw 404 page)
+                 */
+                if ( isset($request->query_vars['category_name']) ) {
+                    $parentCategory = get_category_by_slug($request->query_vars['category_name']);
+                    if (isset($parentCategory->parent) && $parentCategory->parent > 0) {
+                        add_action('wp',   function() {
+                            global $wp_query;
+                            $wp_query->is_404 = true;
```

Comment:
@lms-it have we talked with seo martin if 404 errors or redirects are better?

Reply:
@Xayer This is actually stated in the task description : 
"Please fix it so the direct URLs are not accessible."
Same behavior in old GDS site. In such case the default WP installation have 404 error as well.